src/UniswapV4Migrator.sol

```diff
@@ -0,0 +1,267 @@
+                    uint8(Actions.MINT_POSITION), uint8(Actions.MINT_POSITION), uint8(Actions.SETTLE_PAIR)
+                );
+            }
+        }
+
+        int24 lowerTick =
+            ((TickMath.MIN_TICK + 1) - poolKey.tickSpacing + 1) / poolKey.tickSpacing * poolKey.tickSpacing;
+        int24 upperTick = (TickMath.MAX_TICK - 1) / poolKey.tickSpacing * poolKey.tickSpacing;
+
+        int24 currentTick = TickMath.getTickAtSqrtPrice(sqrtPriceX96);
+
+        if (currentTick < lowerTick || currentTick > upperTick) revert TickOutOfRange();
+
+        liquidity = LiquidityAmounts.getLiquidityForAmounts(
+            sqrtPriceX96,
+            TickMath.getSqrtPriceAtTick(lowerTick),
+            TickMath.getSqrtPriceAtTick(upperTick),
+            uint128(balance0),
+            uint128(balance1)
+        );
+
+        if (liquidity == 0) revert ZeroLiquidity();
+
+        if (isNoOpGovernance) {
+            // For no-op governance, send all liquidity to the protocol locker
+            params[0] = abi.encode(
+                poolKey,
+                lowerTick,
+                upperTick,
+                uint128(liquidity),
+                uint128(balance0),
+                uint128(balance1),
+                address(this),
+                new bytes(0)
+            );
+
+            params[1] = abi.encode(poolKey.currency0, poolKey.currency1);
+        } else {
+            // Standard case: split liquidity 10/90
+            uint256 lockedLiquidity = liquidity / 10;
+            uint256 timeLockLiquidity = liquidity - lockedLiquidity;
+
+            // Liquidity for the protocol locker
+            params[0] = abi.encode(
+                poolKey,
+                lowerTick,
+                upperTick,
+                uint128(lockedLiquidity),
+                uint128(balance0) / 10,
+                uint128(balance1) / 10,
+                address(this),
+                new bytes(0)
+            );
+
+            // Liquidity for the Timelock
+            params[1] = abi.encode(
+                poolKey,
+                lowerTick,
+                upperTick,
+                uint128(timeLockLiquidity),
+                uint128(balance0),
+                uint128(balance1),
+                recipient,
+                new bytes(0)
+            );
+
+            params[2] = abi.encode(poolKey.currency0, poolKey.currency1);
+        }
+
+        if (token0 != address(0)) {
+            ERC20(token0).approve(address(positionManager.permit2()), balance0);
+            positionManager.permit2().approve(token0, address(positionManager), uint160(balance0), type(uint48).max);
+        }
+
+        ERC20(token1).approve(address(positionManager.permit2()), balance1);
+        positionManager.permit2().approve(token1, address(positionManager), uint160(balance1), type(uint48).max);
+
+        positionManager.modifyLiquidities{ value: token0 == address(0) ? balance0 : 0 }(
+            abi.encode(actions, params), block.timestamp
+        );
```

Comment:
By providing `block.timestamp` as the `deadline` here, we're effectively setting an infinite deadline. Normally we would want to add a parameter to this function to be able to pass a `deadline`, but this would require modifying `Airlock`

I think there's no risk here though if we prevent initializing the pool via that hook mechanism, so that's probably the best solution imo